scripts/sync_staging_table_with_production.py

```diff
@@ -0,0 +1,78 @@
+"""
+This script syncs specific tables from the production database to the staging database in Synapse.
+
+Usage: python sync_staging_table_with_prod.py
+"""
+
+from datetime import date
+
+import synapseclient
+from genie import load, process_functions
+
+syn = synapseclient.login()
+syn.table_query_timeout = 50000
+
+tables_to_copy = [
+    "vcf2maf",
+    "bed",
+    "seg",
+    "sample",
+    "patient",
+    "mutationsInCis",
+    "sampleRetraction",
+    "patientRetraction",
+    "sv",
+    "assayinfo",
+    "validationStatus",
+    "errorTracker"
+]
+
+db_to_synid_mapping = {
+    "production":"syn10967259",
+    "staging":"syn12094210"
+}
+
+def download_table(table_key):
+    # download production tables
+    db_table = syn.tableQuery(f"SELECT * FROM {db_to_synid_mapping['production']}").asDataFrame()
+    syn_id = db_table[db_table["Database"] == table_key].Id.values[0]
+    data = syn.tableQuery(f"SELECT * FROM {syn_id}").asDataFrame()
+    return(data)
+    
+
+def replace_table(data_to_replace_with, table_key):
+    # replace staging tables with production tables
+    table_to_replace = syn.tableQuery(f"SELECT * FROM {db_to_synid_mapping['staging']}").asDataFrame()
+    table_to_replace_syn_id = table_to_replace[table_to_replace["Database"] == table_key].Id.values[0]
+    data_to_replace = syn.tableQuery(f"SELECT * FROM {table_to_replace_syn_id}").asDataFrame()
+    
+    # create new table for maf tables
+    if table_key == "vcf2maf":
+        today = date.today()
+        table_name = f"Narrow MAF Database - {today}"
+        new_tables = process_functions.create_new_fileformat_table(
+            syn, 
+            file_format="vcf2maf", 
+            newdb_name=table_name, 
+            projectid="syn22033066",  
+            archive_projectid="syn22033066"
+        )
+        syn.setPermissions(new_tables["newdb_ent"].id, 3326313, [])
+        table_to_replace_syn_id = new_tables["newdb_ent"].id
+        syn.store(synapseclient.Table(table_to_replace_syn_id, data_to_replace_with))
+    else:
+        databaseEnt = syn.get(table_to_replace_syn_id)
+        primary_key = databaseEnt.primaryKey if table_key not in ["validationStatus", "errorTracker"] else ["id"]
+        load._update_table(
```

Comment:
Something I've been thinking about if it just makes sense to completely delete the rows (not delete the table) and then just upload the new rows from prod. This way we are not left with "extra" rows from a previous run of staging and gives us more 1:1 with the production table when we do a staging run. 

So we just add something like:
```
syn.delete(synapseclient.Table(table_to_replace_syn_id, data_to_replace))
```
before the `load._update_table` call

Reply:
This works. In the current script, I didn't see the code to delete the table. It seems [appending](https://github.com/Sage-Bionetworks/Genie/blob/e0faeaa2bc89c853cab83522aa9af31c735c1542/genie/load.py#L336C8-L341C14) rows to be deleted to the dataframe that will be saved to the table. Also, I feel like since we were updagrading to synapseclient 4.9, we might be able to utilize the upsert function of it.

Reply:
That sounds good to me, I think we would still need to do `syn.delete` and then upsert. I initially chose to use `load._update_table` because it was tailored specifically to our genie tables but if we can use [upsert_rows](https://python-docs.synapse.org/en/stable/reference/experimental/sync/table/?h=upsert#synapseclient.models.Table.upsert_rows) here directly, that would be great!

Reply:
Just a precautionary tale - there is a cost to deleting and appending to a table. The table in the Synapse backend still counts all the rows that were added in perpetuity. 

I believe we have encountered such scenarios in genie that caused the platform to have to do optimization upgrades so maybe it is ok to do here. I would advise against doing that with the maf table though